src/tests/generate_trigger_test.ts

```diff
@@ -0,0 +1,61 @@
+import { generateTriggerFile } from "../generate_trigger.ts";
+import { assert, assertEquals } from "../dev_deps.ts";
+
+Deno.test("generateTriggerFile creates a valid TS file", async () => {
+  const input: string[] = [`--source={ "triggers": \
+    [{ "type": "shortcut", \
+    "name": "test trigger", \
+    "description": "Testing file generation", \
+    "workflow": {"callback_id": "test_trigger"}, \
+     "inputs":{ \
+      "interaction": {"value":"{{data.interactivity}}"}
+      } 
+    }]}`];
+
+  const result = await generateTriggerFile(input);
+  const file = await import(`file://${Deno.cwd() + "/" + result.files[0]}`);
+
+  // Directory exists
+  const dirInfo = await Deno.stat("triggers");
```

Comment:
Here, instead of actually interacting with the filesystem (by creating directories), we can instead stub out the `Deno.mkdir` function to prevent it from actually making any changes in the filesystem. This is useful in unit testing (stubbing or mocking out dependencies of the code we are testing in order to isolate and exercise behaviour of specific code).

[Check out the deno.land manual's section on stubbing](https://deno.land/manual@v1.26.2/testing/mocking#stubbing) for an example.

For a rough guide, I think we want to do something like, `let mkdirStub = stub(Deno, 'mkdir');` followed by an assertion that the `mkdirStub` was called with the `"triggers"` string as an argument. That would be sufficient for our purposes for ensuring that a directory was created (we will assume that Deno's stdlib methods like `mkdir` work as advertised).

Reply:
Fair enough; what about validating that the created file would be valid TS? Can the string value written to file be imported with `await import`?